### DISCUSSION TITLE
How to test auto-instrumented code

### DISCUSSION THREAD

**Discussion by rkennedy-mode** [open-telemetry/opentelemetry-java-instrumentation#4550] [Q&A]
We've created an OpenTelemetry extension for JUnit 5, which allows us to capture logged spans. We utilize this to add some additional checking to verify that we're generating the spans we expect with proper parenting. We also run the auto-instrumentation agent in production, which causes some issues for the tests, which we don't run with the agent.

I'm curious if there's any recommendations here regarding running test with/without the agent or some other way to ensure that tests have something approaching what we'll see in production. In particular, we have a fair bit of asynchronous code that jumps across CompletableFuture/ExecutorService type boundaries that we want to make sure are generating spans with the proper parents.

**Comment by trask** [comment]
Created issue to for this #4560
